### PR TITLE
Add HTTP Methods

### PR DESCRIPTION
**Pull request by vitaly-pushkar**
Implementation for #2 
Created it as a Sum type. Not entirely sure if it's a good solution.

### CODE REVIEW

**Comment by vitaly-pushkar** [comment]
@ioleo ✅

**Comment by fsvehla** [comment]
whynotboth.jpeg

```scala
sealed abstract class HttpMethod(string: String)

object HttpMethod {
  final case object GET  extends HttpMethods("GET")
  final case object POST extends HttpMethods("POST")
  // ...
}
```

File: core/src/main/scala/zio/http/Method.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+package zio.http
+
+case class Method(name: String) extends AnyVal {
+  def toString: String = name
+}
+
+// https://tools.ietf.org/html/rfc7231#section-4
```
Comment (jczuchnowski) [inline core/src/main/scala/zio/http/Method.scala]: I don't think we need these links in the code.

**Comment by jczuchnowski** [comment]
I like @fsvehla idea.

**Comment by vitaly-pushkar** [comment]
Thank you @fsvehla @jczuchnowski 
I learned a lot!

**Pull request merged**